docs/src/references/mqtt-topic-guidelines.md

```diff
@@ -119,57 +139,62 @@ This section is divided into 3 parts:
+1. A separate topic structure to exchange data with services running on a device (main or child)
+   which is different from the topics of the device itself.
+1. Support nested child devices so that telemetry data and commands can be exchanged with child devices of child devices.
    deployments with at least 3 levels of nesting: a thin-edge device, its children and grand-children are not uncommon.
-1. When device IDs are used in topics, it is expected that all nested child devices under a thin-edge device are unique.
-   They need not be globally unique or even unique across multiple thin-edge devices.
-1. If child devices can not ensure uniqueness in their IDs,
-   a registration step with thin-edge can be used to get unique IDs under the thin-edge namespace.
-   This registration step must be optional for devices with unique IDs.
+1. A registration service provided by thin-edge to provide unique IDs for requesting child devices.
+   This registration step must be optional for devices if they can guarantee unique IDs on their own.
 1. Support the following kinds of filtering with minimal effort (ideally with a single wildcard subscription):
    * All data from thin-edge device excluding everything else(other services and child devices)
    * All data from all child devices excluding those from thin-edge device and services
    * All data from all services excluding those from thin-edge device and child devices
    * All data from a specific child device excluding everything else
    * All data from a specific service excluding everything else
 1. The topic structure should be ACL friendly so that rules can be applied
-   to limit child devices and services to access only the data meant for them.
+   to limit devices and services to access only the data meant for them.
 1. Enable easier extension of topics with further topic suffixes in future:
    E.g: Support `type` in the topic for measurements like `tedge/measurements/{measurement-type}`
+   or config-type in the topic for config management command: `tedge/commands/req/config_update/<config-type>`
+1. Enable exchange of telemetry data just on the local bus, without it getting forwarded to the cloud.
+1. Thin-edge must be able to generate unique `id`s for child devices on the local network,
+   without consulting with the cloud.
+   The device registration service must work even if the tedge device is not connected to a cloud.
+   All translation between tedge-local-ids and cloud-twin-ids should be done internally by the cloud mapper.
+   It is okay for the mapper to use the same tedge-local-ids in the cloud, if they are unique enough.
+   But, the cloud IDs (typically very long and cumbersome) must not be exposed to the tedge network at any cost.
 
 ### Nice-to have
 
+1. Avoid using the device id in the topics for the main tedge device to keep those well-known topics simple.
+   Use aliases like `main` or `master` instead, if an identifier is really required.
```

Comment:
As said by this comment https://github.com/thin-edge/thin-edge.io/pull/2030#discussion_r1233703411, this is more than keeping topics simple. This is to be able to send message on behalf of the device without having to know the name of that device.

I also see this as related to this comment https://github.com/thin-edge/thin-edge.io/pull/2030#discussion_r1233710365 on the ability to use an MQTT bridge to connect as a child device a device that is running tedge.

Minor: what about using a dot `.` instead of `main`?

Reply:
How about using a `thin-edge` or a `parent` device? I feel that using a special char will not be that meaningful and not very comfortable for readability.

Reply:
I would avoid using `$` in the topic name as it can be confusing when using cli tools as it would require users to use single quotes around the topic instead of double or no quotes to avoid being expanded as a variable.

Below shows an example if we were to use `$MAIN` to represent the main device. Without single quotes, the shell interprets `$MAIN` as a variable and will expand it, resulting in unexpected topic, `tedge//measurements`.

```sh
 mosquitto_pub -t tedge/$MAIN/measurements -m '{}' -d
Client null sending CONNECT
Client null received CONNACK (0)
Client null sending PUBLISH (d0, q0, r0, m1, 'tedge//measurements', ... (2 bytes))
Client null sending DISCONNECT
```

Though it could technically be limited to a single `$` character to represent the **main** device as this would avoid the shell interpreting `$` as a variable, however I think it would not be so obvious to users.

Reply:
@PradeepKiruvale I wouldn't use `parent` for a tedge device because, in the case of hierarchical child devices, there are parents at every level. So, if we're going with words, I'd still go with something like `main` or `root` over `parent`.

@reubenmiller So, are you suggesting that you also prefer `.` over `$` or something more explicit like `main`?

Reply:
The `root` makes sense as well.

Reply:
We decided to:
- keep the term `main` in the specific context of the main device: `te/device/main`
- use an *empty* placeholder when nothing applies as in `te/device/main///`meaning this entity/component is related to a plain entity and not one of its component.